src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonWebToken.PayloadClaimSet.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using Microsoft.IdentityModel.Logging;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.IdentityModel.Tokens.Json;
+
+namespace Microsoft.IdentityModel.JsonWebTokens
+{
+    public partial class JsonWebToken
+    {
+        internal JsonClaimSet CreatePayloadClaimSet(byte[] bytes, int length)
+        {
+            Utf8JsonReader reader = new(bytes.AsSpan().Slice(0, length));
+            if (!JsonSerializerPrimitives.IsReaderAtTokenType(ref reader, JsonTokenType.StartObject, false))
+                throw LogHelper.LogExceptionMessage(
```

Comment:
> throw LogHelper.LogExceptionMessage [](http://example.com/codeflow?start=15&length=36)

do we have a test checking the exception?

Reply:
yes